src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java

```diff
@@ -224,23 +232,51 @@ private static void validateTableAccess(Client engineClient, JSONObject mapping,
         
         String database = mapping.getString("db");
         String table = mapping.getString("table");
+        String format = mapping.optString("format");
+        String mappingName = mapping.optString("mapping");
+        boolean hasAccess = false;
         try {
+            try {
+                KustoOperationResult rs = engineClient.execute(database, String.format(FETCH_TABLE_QUERY, table));
```

Comment:
Since most exceptions are blanketed under the DataService or DataClient exception, we had used a general exception message. we can add an additional error log that surfaces the cause of the exception thrown.

Reply:
what i mean is that in this flow a user that has enough permissions to ingest (table ingestor) can fail this validation as he might not have permissions to query , therefore he will fail this validation.

Reply:
you havent changed that yet ...
please try and use an application with only tableIngestor and make sure it can go through this flow